Dockerfile.base

```diff
@@ -12,11 +12,19 @@
 # can be updated and tested in a separate PR without risking breaking open pull
 # request builds.
 
-FROM ubuntu:24.04
+FROM ubuntu:23.10
 
 # Update and install libraries and tools from repositories
-RUN apt-get update && apt-get dist-upgrade --yes && \
+# Force update glibc and PAM to fix CVE vulnerabilities
+RUN apt-get update && \
+    apt-get dist-upgrade --yes && \
     apt-get install --no-install-recommends --yes \
+      libc-bin \
+      libc6 \
+      libpam-modules \
+      libpam-modules-bin \
+      libpam-runtime \
+      libpam0g \
```

Comment:
Thanks @aurianer for keeping an eye on this. Do you think we could just bump the base image to 24.04 instead?

IIRC the build job is likely failing because we have the credentials only avaialable on pika-org, so you may have to reopen the PR from a new branch...

Also: don't forget to change the version number.